WebAPI/Controllers/StudentController.cs

```diff
@@ -83,6 +85,101 @@ public async Task<IActionResult> GetStudentTimetable(string userEmail)
             return Ok(timetable);
         }
 
+        [HttpGet("getStudentTimetableCalendar/{userEmail}")]
+        public async Task<IActionResult> GetStudentTimetableCalendar(string userEmail)
```

Comment:
It is a big logic inside controller it  should be in some service...

Reply:
I will move the logic

Reply:
moved to calendar service